File: PeduliPanti - BE/app/Models/RequestList.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class RequestList extends Model
{
    use HasFactory;

    protected $table = 'request_list';
    protected $primaryKey = 'requestID';
    public $timestamps = false;

    protected $fillable = [
        'pantiID',
        'productID',
        'requested_qty',
        'donated_qty',
        'status_approval',
    ];

    public function panti()
    {
        return $this->belongsTo(PantiDetail::class, 'pantiID', 'pantiID');
    }

    public function products()
    {
        return $this->belongsTo(Product::class, 'requestID', 'productID');
    }
    
}

File: PeduliPanti - FE/lib/Models/RequestList.dart
class RequestList {
  final int id;
  final int pantiID;
  final int productID;
  final int requestedQty;
  final int donatedQty;
  final String statusApproval;

  RequestList({
    required this.id,
    required this.pantiID,
    required this.productID,
    required this.requestedQty,
    required this.donatedQty,
    required this.statusApproval,
  });

  factory RequestList.fromJson(Map<String, dynamic> json) {
    return RequestList(
      id: json['id'] ?? 0,
      pantiID: json['pantiID'] ?? 0,
      productID: json['productID'] ?? 0,
      requestedQty: json['requested_qty'] ?? 0,
      donatedQty: json['donated_qty'] ?? 0,
      statusApproval: json['status_approval'] ?? 'rejected',
    );
  }

  RequestList copyWith({
    int? id,
    int? pantiID,
    int? productID,
    int? requestedQty,
    int? donatedQty,
    String? statusApproval,
  }) {
    return RequestList(
      id: id ?? this.id,
      pantiID: pantiID ?? this.pantiID,
      productID: productID ?? this.productID,
      requestedQty: requestedQty ?? this.requestedQty,
      donatedQty: donatedQty ?? this.donatedQty,
      statusApproval: statusApproval ?? this.statusApproval,
    );
  }
}
